src/clients/java/src/client.zig

```diff
@@ -111,27 +109,26 @@ const NativeClient = struct {
             return undefined;
         };
 
-        var out_packet: ?*tb.tb_packet_t = null;
-        const acquire_status = tb.acquire_packet(context.client, &out_packet);
-
-        if (out_packet) |packet| {
-            assert(acquire_status == .ok);
-
-            // Holds a global reference to prevent GC before the callback.
-            const global_ref = JNIHelper.new_global_reference(env, request_obj);
-
-            packet.operation = operation;
-            packet.user_data = global_ref;
-            packet.data = send_buffer.ptr;
-            packet.data_size = @intCast(send_buffer.len);
-            packet.next = null;
-            packet.status = .ok;
-            tb.submit(context.client, packet);
-        } else {
-            assert(acquire_status != .ok);
-        }
-
-        return acquire_status;
+        // Compute a valid, aligned Packet pointer inside packet_buffer.
+        const packet_buf: []u8 = ReflectionHelper.get_packet_buffer_slice(env, request_obj) orelse
+            @panic("Request.packetBuffer somehow became null or invalid");
+        const packet_align_ptr =
+            std.mem.alignForward(usize, @intFromPtr(packet_buf.ptr), @alignOf(tb.tb_packet_t));
+        assert(packet_align_ptr + @sizeOf(tb.tb_packet_t) <=
+            @intFromPtr(packet_buf.ptr + packet_buf.len));
```

Comment:
Sorry for going back and forth with this, but it seems too complicated (and adds alignment overhead!).
We could use [`java.lang.foreing`](https://docs.oracle.com/en%2Fjava%2Fjavase%2F22%2Fdocs%2Fapi%2F%2F/java.base/java/lang/foreign/package-summary.html) instead, but it's only available on Java 22.

I wonder if a plain `allocator.create(tb.tb_packet_t)` would be just simpler here (and with the same effect allocating off-heap memory) :smiling_face_with_tear: